### PR TITLE
🔥 withdraw on burn item

### LINKED ISSUE #3584: Burning NFT should cancel Offers
Expanding from @roileo comment here:

why do we need to show burned content?
If it's burned, the offer shouldn't exist anymore

_Originally posted by @roiLeo in https://github.com/kodadot/nft-gallery/pull/3575#pullrequestreview-1051999846_

Currently, burning NFT doesn't affect offers around the NFT at all. I suppose, this is not ideal since it would be only logical to have burning NFT with offers trigger withdrawal / cancellation of offers. 

@vikiival wdyt, what would it take to have it working like this

### PR DESCRIPTION
**Pull request by roiLeo**
Smelly code :/ need some refactoring in `transactionConsume` watcher

## PR Type

- [x] Feature

## Context

- [x] Closes #3584
- [x] Related #4592 
- [x] Requires #4771 

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [x] I've tested it at `/snek/gallery/659233203-2`

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=HguGafrfk8nRP9a4tgJf7hLvSSFie8GtkDjyWFHQ2kDUAMc)

## Screenshot 📸

https://user-images.githubusercontent.com/9987732/214058571-6e8ef1ec-74cf-43da-99cd-557cc9acc5f1.mov

### CODE REVIEW

**Comment by vikiival** [comment]
Am I blind or where is the burn button?

**Comment by roiLeo** [comment]
> Am I blind or where is the burn button?

https://github.com/kodadot/nft-gallery/issues/4763#issuecomment-1396943625

"More actions" > "Burn"

File: composables/transaction/transactionConsume.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          collectionId,
+          tokenId,
+          offer.caller
+        )
+      })
+      const cb = hasOffers.value
+        ? api.tx.utility.batchAll
+        : getApiCall(api, item.urlPrefix, Interaction.CONSUME)
+
+      console.log([offerWithdrawArgs])
```
[the remark this answers is not in the document]
Comment (vikiival) [reply composables/transaction/transactionConsume.ts:44]: 🔊

File: composables/transaction/transactionConsume.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import { Interaction, createInteraction } from '@kodadot1/minimark'
+
+import { bsxParamResolver, getApiCall } from '@/utils/gallery/abstractCalls'
+import type { ActionConsume } from './types'
+
+export function exectConsumeTx(item: ActionConsume, api, executeTransaction) {
```
[the remark this answers is not in the document]
Comment (roiLeo) [reply composables/transaction/transactionConsume.ts]: sloth

File: composables/transaction/transactionBurn.ts [hunk 3]
```diff
@@ -14,11 +14,42 @@ export function execBurnTx(item: ActionConsume, api, executeTransaction) {
...
+      const arg = [
+        [...offerWithdrawArgs, api.tx.nft.burn(collectionId, tokenId)],
+      ]
+
+      executeTransaction({
+        cb: cb,
+        arg: arg,
+        successMessage: item.successMessage,
+        errorMessage: item.errorMessage,
+      })
```
Comment (vikiival) [inline composables/transaction/transactionBurn.ts:52]: Missing `arg: bsxParamResolver(item.nftId, Interaction.CONSUME, ''),`

**Review by vikiival — CHANGES_REQUESTED** [review]
please clarify

File: composables/transaction/transactionBurn.ts (hunk 3 above)
Comment (roiLeo) [reply composables/transaction/transactionBurn.ts:52]: > Missing `arg: bsxParamResolver(item.nftId, Interaction.CONSUME, ''),`

same thing as `api.tx.nft.burn(collectionId, tokenId)`

File: composables/transaction/transactionBurn.ts [hunk 4]
```diff
@@ -28,15 +28,17 @@ export function execBurnTx(item: ActionConsume, api, executeTransaction) {
...
+          return api.tx.marketplace.withdrawOffer(
+            collectionId,
+            tokenId,
+            offer.caller
+          )
+        }
+      )
       const cb = hasOffers.value
         ? api.tx.utility.batchAll
         : getApiCall(api, item.urlPrefix, Interaction.CONSUME)
```
Comment (vikiival) [inline composables/transaction/transactionBurn.ts:44]:
```suggestion
      const cb = api.tx.utility.batchAll
```

**Review by vikiival — CHANGES_REQUESTED** [review]
TO make it work with current impl

File: composables/transaction/transactionBurn.ts (hunk 3 above)
Comment (vikiival) [reply composables/transaction/transactionBurn.ts]: Not the same sir

File: composables/transaction/transactionBurn.ts (hunk 4 above)
Comment (roiLeo) [reply composables/transaction/transactionBurn.ts]: oops my bad I've missed it

**Review by vikiival — APPROVED** [review]

**Comment by yangwao** [comment]
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 36.74 USD/KSM ~ 1.361 $KSM
🧗 [HguGafrfk8nRP9a4tgJf7hLvSSFie8GtkDjyWFHQ2kDUAMc](https://kusama.subscan.io/account/HguGafrfk8nRP9a4tgJf7hLvSSFie8GtkDjyWFHQ2kDUAMc)
🔗 [0xd035fe74d9622fd5a55f7396b85a7ac490516f4c75bd6944f64197e4a69f1df9](https://kusama.subscan.io/extrinsic/0xd035fe74d9622fd5a55f7396b85a7ac490516f4c75bd6944f64197e4a69f1df9)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**